joomla/joomla-cms
# [CS] Fixing JImage constants

### Summary of Changes
1) Switched from hardcoded variables to class constants when checking for crop type
2) Using `static` instead of `self` in methods

This is a coding style change and doesn't have any effects on functionality.
Only change would if one would extend the `JImage` class: developer doesn't have to define JImage constants again in extended class.

### Testing Instructions
The only place where _JImage_ is used in J! core is template editing feature:
1. Go to _Extensions > Templates > Templates > Protostar Details and Files_
2. Click on _template_preview.png_
3. Make sure cropping and resizing works as expected

### Expected result
Unit suites should pass
Manual tests pass

### Actual result
same as expected result


### Documentation Changes Required
No changes required
